programs/utilities/content_node.py

```diff
@@ -0,0 +1,296 @@
+        return soup.get_text(separator='\n').strip()
+
+
+    def __set_node_details(self):
+        """
+        Routing function. Based on what type of node we're
+        dealing with, further type specific processing
+        will happen.
+        """
+        if self.node_type == ContentNodeType.TITLE:
+            self.__title_processing()
+        elif self.node_type == ContentNodeType.LIST:
+            self.__list_processing()
+        elif self.node_type == ContentNodeType.LIST_ITEM:
+            self.__list_item_processing()
+        elif self.node_type == ContentNodeType.CONTENT:
+            self.__content_processing()
+        else:
+            return
+
+    def __title_processing(self):
+        """
+        Processes nodes determined to be TITLEs. We currently
+        don't send these off to comprehend for processing and
+        instead do simple string lookups.
+        """
+        cleaned_line = self.cleaned.strip()
+        self.lines.append({
+            'text': cleaned_line,
+            'personal_data': [],
+            'entities': []
+        })
+
+        if any([x in cleaned_line.lower() for x in ['application', 'admission']]):
+            self.content_category = ContentCategory.ADMISSIONS
+        elif any([x in cleaned_line.lower() for x in ['curriculum', 'credit hour', 'electives']]):
+            self.content_category = ContentCategory.COURSES
+        elif any([x in cleaned_line.lower() for x in ['contact', 'address', 'phone']]):
+            self.content_category = ContentCategory.CONTACT_INFO
+        else:
+            self.content_category = ContentCategory.UNKNOWN
+
+
+    def __list_processing(self):
+        """
+        Processes nodes determined to be LISTs. We currently
+        don't send these off to comprehend for processing and
+        instead do some regex lookups to determine if they
+        contain progam course information.
+        """
+        course_re = re.compile(r'([A-Za-z]{3,4}\s)([0-9]{4})')
+
+        cleaned_lines = self.cleaned.splitlines()
+        course_line_score = 0
+
+        for line in cleaned_lines:
+            result = course_re.match(line)
+            if result:
+                course_line_score += 100
+
+        avg_score = course_line_score / len(cleaned_lines)
+
+        if avg_score > 80:
+            self.content_category = ContentCategory.COURSES
+        else:
+            self.content_category = ContentCategory.GENERAL
+
+    def __list_item_processing(self):
+        """
+        Not used. TODO: Consider removing.
+        """
+        pass
+
+    def __content_processing(self):
+        """
+        Processes nodes determined to be CONTENT types. We
+        split each separate line of content and send it off
+        to AWS Comprehend for language processing.
+        """
+        cleaned_lines = self.cleaned.splitlines()
+
+        contact_info_score = 0
+        contact_re = re.compile(r'^(?:College of|Department of) [a-zA-Z\ ]+$')
```

Comment:
We may need to update the "College of" portion to optionally include "Rosen" before it to account for hospitality programs.